messaging/src/main/java/org/axonframework/eventhandling/processors/streaming/pooled/SplitTask.java

```diff
@@ -109,7 +109,7 @@ private CompletableFuture<Boolean> fetchSegmentAndSplit(int segmentId) {
         return unitOfWorkFactory
                 .create()
                 .executeWithResult(context -> {
-                    int[] segments = tokenStore.fetchSegments(name);
+                    int[] segments = joinAndUnwrap(tokenStore.fetchSegments(name, null));
```

Comment:
```suggestion
                    int[] segments = joinAndUnwrap(tokenStore.fetchSegments(name, context));
```

We should pass the `ProcessingContext` here.